util/rate_limiter_test.cc

```diff
@@ -416,6 +416,37 @@ TEST_F(RateLimiterTest, LimitChangeTest) {
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
 }
 
+TEST_F(RateLimiterTest, AvailableByteSizeExhaustTest) {
+  SpecialEnv special_env(Env::Default(), /*time_elapse_only_sleep*/ true);
+  const std::chrono::seconds kTimePerRefill(1);
+
+  std::shared_ptr<RateLimiter> limiter = std::make_shared<GenericRateLimiter>(
+      500 /* bytes */, std::chrono::microseconds(kTimePerRefill).count(),
+      10 /* fairness */, RateLimiter::Mode::kWritesOnly,
+      special_env.GetSystemClock(), false /* auto_tuned */);
+
+  limiter->Request(100, Env::IO_USER, nullptr /* stats */,
+                   RateLimiter::OpType::kWrite);
+  special_env.SleepForMicroseconds(
+      static_cast<int>(std::chrono::microseconds(kTimePerRefill).count()));
+
+  ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->SetCallBack(
+      "GenericRateLimiter::Request:PostEnqueueRequest", [&](void* arg) {
+        port::Mutex* request_mutex = (port::Mutex*)arg;
+        request_mutex->Unlock();
+        ASSERT_EQ(500, limiter->GetTotalBytesThrough(Env::IO_USER));
```

Comment:
nit: will appreciate a one-sentence summary of what assert_eq(500...) means for this test.